File: exercicios/java/Exercicio05/src/main/java/org/example/Funcionario.java
package org.example;

public abstract class Funcionario extends UsuarioBU {
    protected String departamento;
    public Funcionario(String departamento, long cpf, int diasDeEmprestimo) {
        super(cpf, diasDeEmprestimo);
        this.departamento = departamento;
    }
    public String getDepartamento() {
        return departamento;
    }
    public void setDepartamento(String departamento) {
        this.departamento = departamento;
    }
    @Override
    public String devolver(String tituloLivro) {
        String devolucaoDoUsuario = super.devolver(tituloLivro);
        return "Funcionário do departamento \"" + this.departamento + devolucaoDoUsuario;
    }
    @Override
    public String emprestar(String tituloLivro) {
        String emprestarDoUsuario = super.emprestar(tituloLivro);
        String strDiasDeEmprestimo = String.valueOf(this.diasDeEmprestimo);
        return "do departamento \"" + this.departamento + emprestarDoUsuario  + strDiasDeEmprestimo + " dias de prazo";
    }
}

File: exercicios/python/Exercicio05/funcionario.py
from abc import ABC, abstractmethod
from usuario_bu import UsuarioBU

class Funcionario(UsuarioBU):

    @abstractmethod
    def __init__ (self, departamento:str, cpf:int, dias_de_emprestimo:int):
        super().__init__(cpf, dias_de_emprestimo)
        self.__departamento = departamento

    @property
    def departamento(self):
        return self.__departamento
    
    @departamento.setter
    def departamento(self, departamento):
        self.__departamento = departamento
    
    def emprestar (self, titulo_do_livro:str):
        return 'do departamento "' + self.__departamento + super().emprestar(titulo_do_livro)  + str(self.dias_de_emprestimo) + ' dias de prazo'

    def devolver (self, titulo_do_livro:str):
        return 'do departamento "' + self.__departamento + super().devolver(titulo_do_livro)
